lrlex/src/lib/parser.rs

```diff
@@ -96,21 +113,27 @@ impl<StorageT: TryFrom<usize>> LexParser<StorageT> {
         let declaration = self.src[i..i + declaration_len].trim_end();
         match declaration {
             "%x" => {
-                for name in self.src[i + declaration_len..i + line_len]
+                let start_states: HashSet<&str> = self.src[i + declaration_len..i + line_len]
                     .trim()
                     .split_whitespace()
-                {
+                    .map(|name| self.validate_start_state(i, name))
+                    .collect::<LexInternalBuildResult<HashSet<&str>>>()?;
+
+                for name in start_states {
                     let id = self.start_states.len();
                     let start_state = StartState::new(id, name, true);
                     self.start_states.push(start_state);
                 }
                 Ok(i + line_len)
             }
-            "%s" => {
-                for name in self.src[i + declaration_len..i + line_len]
+            _ if (declaration.to_lowercase()=="%s" || declaration.to_lowercase()=="%start") => {
```

Comment:
I *think* the other declarations are case sensitive (or we treat them as such). Does lex treat them case sensitive/insensitive?

Reply:
According to https://pubs.opengroup.org/onlinepubs/007904875/utilities/lex.html#tag_04_73_13_01

> ...any line beginning with a '%' (percent sign) character and followed by an alphanumeric word beginning with either 's' or 'S' shall define a set of start conditions. Any line beginning with a '%' followed by a word beginning with either 'x' or 'X' shall define a set of exclusive start conditions. When the generated scanner is in a %s state, patterns with no state specified shall be also active; in a %x state, such patterns shall not be active. The rest of the line, after the first word, shall be considered to be one or more <blank>-separated names of start conditions. 

I hadn't seen this before - if you agree, I'll adjust the logic to match?

Reply:
Works for me!

Reply:
Updated in https://github.com/softdevteam/grmtools/pull/318/commits/c91dab0b6433d3650ab08dae6172849d7d569444

Reply:
Looks good, but could we add a negative test or two for names that shouldn't be accepted? That'll stop us accidentally mucking this up in the future!

Reply:
See tests in c2348a3a6118a0c797d7202a7b9de9b578b8229c